app/javascript/app/components/sectors-agriculture/drivers-of-emissions/historical-emissions-graph/historical-emissions-graph-selectors.js

```diff
@@ -0,0 +1,298 @@
+import { createSelector, createStructuredSelector } from 'reselect';
+import qs from 'query-string';
+import { uniqBy } from 'lodash';
+import {
+  getYColumnValue,
+  getThemeConfig,
+  getTooltipConfig,
+  setChartColors,
+  getColumnValue,
+  getPieChartThemeConfig
+} from 'utils/graphs';
+import {
+  CHART_COLORS,
+  CHART_COLORS_EXTENDED,
+  GREY_CHART_COLORS,
+  DEFAULT_AXES_CONFIG
+} from 'data/constants';
+import { format } from 'd3-format';
+
+const getSourceSelection = state =>
+  (state.location && state.location.search) || null;
+const getCountries = state => state.countries || null;
+const getRegions = state => state.regions || null;
+const getAgricultureEmissionsData = state =>
+  (state.agricultureEmissions && state.agricultureEmissions.data) || null;
+const getAgricultureEmissionsLoading = state =>
+  (state.agricultureEmissions && state.agricultureEmissions.loading) || false;
+const getGhgEmissionsData = state =>
+  (state.ghgEmissions && state.ghgEmissions.data) || null;
+const getGhgEmissionsLoading = state =>
+  (state.ghgEmissions && state.ghgEmissions.loading) || false;
+
+const API_SCALE = 0.001; // converting from Gigagrams to Megatonnes ( 1 Gg = 0.001 Mt)
```

Comment:
thanks for the comment, it's very useful actually, I'm gonna start doing that too :D 💃

Reply:
I hope it's correct :D